lib/mix/nerves/preflight.ex

```diff
@@ -40,33 +40,32 @@ defmodule Mix.Nerves.Preflight do
     ensure_available!("mksquashfs", package: "squashfs")
   end
 
+  @spec ensure_fwup_version!(String.t(), String.t()) :: :ok
   def ensure_fwup_version!(fwup_bin \\ "fwup", vsn_requirement \\ @fwup_semver) do
     ensure_available!(fwup_bin)
 
-    with {vsn, 0} <- Nerves.Port.cmd(fwup_bin, ["--version"]),
-         vsn = String.trim(vsn),
-         {:ok, req} = Version.parse_requirement(vsn_requirement),
-         true <- Version.match?(vsn, req) do
-      :ok
-    else
-      false ->
-        {vsn, 0} = Nerves.Port.cmd(fwup_bin, ["--version"])
+    {vsn, 0} = Nerves.Port.cmd(fwup_bin, ["--version"])
+    vsn = String.trim(vsn)
+    {:ok, req} = Version.parse_requirement(vsn_requirement)
 
-        Mix.raise("""
-        #{fwup_bin} #{vsn_requirement} is required for Nerves.
+    unless Version.match?(vsn, req) do
+      Mix.raise("""
+      #{fwup_bin} #{vsn_requirement} is required for Nerves.
 
-        You are running #{vsn}.
-        Please see https://hexdocs.pm/nerves/installation.html#fwup
-        for installation instructions
-        """)
-
-      error ->
-        Mix.raise("""
-        Nerves encountered an error while checking host requirements for fwup
-        #{inspect(error)}
-        Please open a bug report for this issue on github.com/nerves-project/nerves
-        """)
+      You are running #{vsn}.
+      Please see https://hexdocs.pm/nerves/installation.html#fwup
+      for installation instructions
+      """)
     end
+
+    :ok
+  rescue
```

Comment:
If we raise above, then we're going to capture it here and then raise again

Reply:
Thanks. I updated with a more minor change that gets past the dialyzer warning.